chainercv/utils/iterator/apply_prediction_to_iterator.py

```diff
@@ -0,0 +1,141 @@
+from chainercv.utils.iterator.unzip import unzip
+
+
+def apply_prediction_to_iterator(predict, iterator, hook=None):
+    """Apply a prediction function/method to an iterator.
+
+    This function applies a prediction function/method to an iterator.
+    It assumes that the iterator returns a batch of images or
+    a batch of tuples whose first element is an image. In the case that
+    it returns a batch of tuples, the rests are treated as ground truth
+    values.
+
+    >>> imgs = next(iterator)
+    >>> # imgs: [img]
+    or
+    >>> batch = next(iterator)
+    >>> # batch: [(img, gt_val0, gt_val1)]
+
+    This function applys :func:`predict` to a batch of images and gets
+    predicted value(s). :func:`predict` should take a batch of images and
+    return a batch of prediction values
+    or a tuple of batches of prediction values.
+
+    >>> pred_vals0 = predict(imgs)
+    >>> # pred_vals0: [pred_val0]
+    or
+    >>> pred_vals0, pred_vals1 = predict(imgs)
+    >>> # pred_vals0: [pred_val0]
+    >>> # pred_vals1: [pred_val1]
+
+    Here is an exmple which applies a pretrained Faster R-CNN to
```

Comment:
If its restrictive, don't we not use commas? 

Also, I think `, which` should be used for this line, and ` that (no comma)` for Args.